### DISCUSSION TITLE
How to remove or hide "create & create another" button from Modal and Create page

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6114] [Help] [answered]
I want to hide `create & create another` button while creating new data, In edit page (or modal) it is not showing, how can I make this button hidden while creating & updating

thanks in advance

**Comment by danharrin** [comment, marked as answer]
It's an option in the config file I think

**Reply by valpuia** [reply]
Cannot find anything related to this data in config file..

**Reply by danharrin** [reply]
On the create page:

```php
protected static bool $canCreateAnother = true;
```

On `CreateAction` modals: `CreateAction::make()->disableCreateAnother()`

**Reply by valpuia** [reply]
It is working.. thanks

**Reply by AnderssonGt** [reply]
Hi!, And for the "close" button, is there a way to hide it too?

**Reply by SourabhaSMohanty** [reply]
yes but it will hide the close button from all the modals

public function getModalCancelAction(): ModalAction
    {
        if ($this->modalCancelAction) {
            return $this->evaluate($this->modalCancelAction);
        }

        return static::makeModalAction('cancel')
            ->label(__('filament-support::actions/modal.actions.cancel.label'))
            ->cancel()
            ->hidden()
            ->color('secondary');
    }
    
 is there any way to hide the button from a perticular modal?

**Reply by Binyamkpaul** [reply]
what about in relationship manager?

**Reply by GhostvOne** [reply]
> what about in relationship manager?

I was looking for the same question,
Found the fix : 
```php
 ->headerActions([
                Tables\Actions\CreateAction::make()
                    ->createAnother(false),
            ])
```

**Comment by ADdora** [comment]
For **Filament 3**
`protected static bool $canCreateAnother = false;` **does not work**

`disableCreateAnother()` **deprecated**

According to [Filament 3 docs](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#disabling-create-another)  ```CreateAction::make()->createAnother(false)``` works with me.

```
    public function table(Table $table): Table
    {
        return $table
            ->columns([
                // ..
            ])
            ->headerActions([
                Tables\Actions\CreateAction::make()
                    ->createAnother(false)  
            ])
            ->actions([
                // ..
            ]);
    }
```
Hope That help.

**Reply by salvationarinze** [reply]
Using `protected static bool $canCreateAnother = false;` in the create page worked for me

**Reply by kaythinks** [reply]
this worked for me as well.

**Comment by easycash01** [comment]
for remove the create button insert this
 
 public static function canCreate(): bool
    {
        return false;
    }

**Comment by wscheah** [comment]
instead of hiding the create & create another, possible to rename it?

**Reply by danharrin** [reply]
Override `getCreateAnotherFormAction()` with `return parent::getCreateAnotherFormAction()->label('new label')`

**Reply by danswiser** [reply]
> Override `getCreateAnotherFormAction()` with `return parent::getCreateAnotherFormAction()->label('new label')`

Is this possible without having a CreateResource page? I'm just using a simple resource with a modal to create the resource and want to change the label.

**Reply by danswiser** [reply]
Never mind. I found it in the translatable files and changed the label there.

`php artisan vendor:publish --tag=filament-actions-translations` and then in `lang\vendor\filament-actions\en\create.php`

**Comment by Ahmed-gubara** [comment]
I used this code to disable the _create & create another_ in the _Create record Page_, and the _Create record Modal_.

```php
class AppServiceProvider extends ServiceProvider
{
    public function boot(): void
    {
        // only add these two lines to the boot() function on the AppServiceProvider class
        \Filament\Resources\Pages\CreateRecord::disableCreateAnother();
        \Filament\Actions\CreateAction::configureUsing(fn(CreateAction $action) => $action->createAnother(false));
    }
```
Maybe it was added by an update.

_I found this discussion from google, adding this code may help the next developer._
